pegasus_wrapper/key_value.py

```diff
@@ -164,6 +164,13 @@ def downsample(
     )
 
 
+@attrs(frozen=True, slots=True)
+class DataSplit:
+    train: KeyValueStore = attrib(validator=instance_of(KeyValueStore), kw_only=True)
+    dev: KeyValueStore = attrib(validator=instance_of(KeyValueStore), kw_only=True)
+    test: KeyValueStore = attrib(validator=instance_of(KeyValueStore), kw_only=True)
```

Comment:
Will it be an issue if any of these ever need to not exist/be empty?

Reply:
I went back and forth on this. On the one hand it is nice to be able to have one or more of these be `Optional`; on the other hand, that pollutes user code with lots of checks for `None` when in context they are quite sure of what their splits are.  I decided on just having an empty `KeyValueStore`s if e.g. there isn't a dev set, but I could be convinced to go the other way.

Reply:
I didn't know if the stores could be empty. In my opinion, if a `KeyValueStore` can be empty in the same way a normal container can, then this is a better solution, it just isn't clear to me from the rest of the file how that works.

Reply:
Does 0a379de7a42da0dde9ac4b469400a3acd79baa73 help?

Reply:
Yes, I think so.